### ISSUE TITLE
Cannot change CanClose dynamically

### ISSUE THREAD

**Issue by messani** [xceedsoftware/wpftoolkit#1646]
I am trying to change visibility of a close button dynamically. I tried to do this by attaching a style to a LayoutItem like this:
```
<Style TargetType="{x:Type xcadctrl:LayoutItem}">
  <Setter Property="CanClose" Value="{Binding Model.CanClose}" />
</Style>
```
CanClose property is bound to my model, but it's value is read only once when the view is created. When I change CanClose property at model later, nothing happens. I tried to debug wpftoolkit and found out that LayoutItem.CanClose is set locally at LayoutItem.SetDefaultBindings()

https://github.com/xceedsoftware/wpftoolkit/blob/d155bd577e5d7b8f2c36468fe9991a244dd7af2e/ExtendedWPFToolkitSolution/Src/Xceed.Wpf.AvalonDock/Controls/LayoutItem.cs#L1186

I removed this assignment and my code worked. IMHO there should not be this assignment.

**Comment by XceedBoucherS** [comment]
Hi,
I can see what you mean.
But can you submit a complete small sample project showing the issue ? Just want to make sure we test the same code as you.
Thank you.

**Comment by messani** [comment]
[XceedBug.zip](https://github.com/xceedsoftware/wpftoolkit/files/6031044/XceedBug.zip)

Hi. This is minimal example. The checkbox should dynamically change visitility of the close button.

I found an another bug (see comments in code - search for word "Messani"). If I create datacontext from XAML (it is commented out), it is not possible to set CanClose property from the ViewModel at all.

**Comment by XceedBoucherS** [comment]
Hi,

In LayoutItem.SetDefaultBindings(), try replacing the :
CanClose = LayoutElement.CanClose;
with:
this.SetCurrentValue( LayoutItem.CanCloseProperty, LayoutElement.CanClose );
This will prevent from overriding a binding.

The same can be done for :
IsSelected = LayoutElement.IsSelected;
IsActive = LayoutElement.IsActive;

This will be fixed in v4.2.

I believe this also fixes the other bug you found with setting the DataContext from XAML.

**Comment by messani** [comment]
I tried it and it is working now.  Thank you.
